### PR TITLE
fix(NetworkRequest): error if bad response or status code

### PR DESCRIPTION
**Pull request by woudini**
## Objective

- Handle errors in a way similar to `KYCNetworkRequest`.
- Make `OrderResult` (the response from the trades endpoint) non-optional.
- Update `OrderResult` properties to match server response.

## Description

- Added error handling to `NetworkRequest`'s `execute` method.
- Removed optionals and error-related properties from `OrderResult`.
- Removed force unwraps of `OrderResult` properties from `TradeExecutionService`.

## How to Test

Hit a bad endpoint and see an error instead of an attempt to decode to a certain type.

## Screenshot/Design assessment

N/A

## Merge Checklist

- [x] The PR uses a title supported by [.changelogrc](https://github.com/blockchain/My-Wallet-V3-iOS/blob/dev/.changelogrc#L6...L69).
- [ ] Areas of technical debt are marked with a `// TICKET:` comment that includes a ticket number.
- [x] All unit tests pass.
- [ ] You have added unit tests.
- [ ] New files added are within the correct directory. (e.g. if a file is required for unit tests to compile, be sure it is added to the tests target.)

### CODE REVIEW

File: Blockchain/Network/Models/Result.swift [hunk 1]
```diff
@@ -16,3 +16,8 @@ enum Result<A> {
     case success(A)
     case error(Error?)
 }
+
+enum HTTPRequestResult<A> {
```
Comment (bchrisarriola) [inline Blockchain/Network/Models/Result.swift]: Not sure I see the advantages of this. Why not just use `Result` and if the case is `error`, check if the value is of type `HTTPRequestError`?

Comment (woudini) [reply Blockchain/Network/Models/Result.swift]: Removed

File: Blockchain/Network/NetworkRequest.swift [hunk 2]
```diff
@@ -77,25 +77,33 @@ struct NetworkRequest {
...
+                withCompletion(.error(HTTPRequestClientError.failedRequest(description: error.localizedDescription)), responseCode)
+                return
+            }
+
+            guard let httpResponse = response as? HTTPURLResponse else {
+                withCompletion(.error(HTTPRequestServerError.badResponse), responseCode)
+                return
+            }
+
+            guard (200...299).contains(httpResponse.statusCode) else {
```
Comment (thisisalexmcgregor) [inline Blockchain/Network/NetworkRequest.swift]: `var resonseCode: Int = 0` on line 68 can be removed

Comment (woudini) [reply Blockchain/Network/NetworkRequest.swift]: Can I change to just `let`? Otherwise I'll be passing in `0` a few times in this function.

Comment (thisisalexmcgregor) [reply Blockchain/Network/NetworkRequest.swift]: Oh yeah, that's what I meant. Whoops.

**Pull request merged**